tests/grid/grid_tools_collect_periodic_faces_parallel_shared_tria.cc

```diff
@@ -0,0 +1,91 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2001 - 2023 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+#include <deal.II/base/exceptions.h>
+#include <deal.II/base/mpi.h>
+#include <deal.II/base/point.h>
+
+#include <deal.II/distributed/shared_tria.h>
+#include <deal.II/distributed/tria.h>
+
+#include <deal.II/grid/grid_generator.h>
+#include <deal.II/grid/grid_tools.h>
+#include <deal.II/grid/tria.h>
+
+#include <vector>
+
+#include "../tests.h"
+
+template <int dim>
+void
+test(MPI_Comm comm)
+{
+  deallog << "dim = " << dim << std::endl;
+  parallel::shared::Triangulation<dim> tria(comm);
+
+  std::vector<typename GridTools::PeriodicFacePair<
+    typename parallel::shared::Triangulation<dim>::cell_iterator>>
+    matched_pairs;
+
+  unsigned int num_refinements = 1 << 4;
+
+  if constexpr (dim == 1)
```

Comment:
Sorry for not catching that earlier but you can't use C++17 features. We are only requiring C++14.

Reply:
Your Godbolt example looks very interesting, even though I do not fully understand. If it is only about the `if constexpr`, then it might be easier to just write a for loop as suggested in my last commit. `GridGenerator` functions and `GridTools` function are "run-time" functions (i.e. there arguments do not have to be known during compilation).

Reply:
I just saw, that I mispelled the variable `repititions`, in the above program it is `repitions`. Should I change it, or just leave it as it is?

Reply:
Yeah, writing the loop explicitly is good.